apis/config/v1beta1/defaults.go

```diff
@@ -60,6 +60,8 @@ func getOperatorNamespace() string {
 }
 
 // SetDefaults_Configuration sets default values for ComponentConfig.
+//
+//nolint:revive // backward compatibility
```

Comment:
I personally don't like the use of magic comments for liners.

Reply:
TBH, I am ok changing the name of these functions.

Reply:
@trasc @alculquicondor IIUC, this is the limitation of kubernetes. So we can not change this name.
Please see `zz_generated.defaults.go` for more details.

Reply:
In that case, the comment should say `// format required by generated code for defaulting`

Reply:
That makes sense.

Reply:
If in the future the linter is dropped , the "special" comment will remain in the code with no use.

Reply:
> In that case, the comment should say // format required by generated code for defaulting

Agree, I will change the comment.

> If in the future the linter is dropped , the "special" comment will remain in the code with no use.

It's possible to use [`nolintlint`](https://golangci-lint.run/usage/linters/#nolintlint) linter to detect unused `//nolint` comments.

Reply:
Kubernetes project [actively uses](https://github.com/search?q=repo%3Akubernetes%2Fkubernetes+%22%2F%2Fnolint%3A%22&type=code) `//nolint` directives. That's why I decided to use `//nolint` comments instead of disabling lint issue via `.golangci.yml` config.

Reply:
I'm ok with the comment. What does `revive` referring to, though?

Reply:
[`revive`](https://golangci-lint.run/usage/linters/#revive) is the linter which detects var-naming issues.